### PR TITLE
[feature] use statusBarError Background Color to match themes

### PR DESCRIPTION
**Pull request by EEParker**
## Background 
This pull requests adds support for the new status bar error background color. This color is defined by the currently selecte theme, and allows the warning to stand out more, especially when debugging.

_**NOTE: this is marked DRAFT as this feature still requires the `proposedApi` flag. I planned on removing the WIP once the NPM for `vscode` is updated.**_

### New:
image
image

### Previous:
image
image

## More details on this feature:
https://github.com/microsoft/vscode/issues/110214

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -47,7 +47,8 @@ function updateConfigs(status:vscode.StatusBarItem) {
     let config = vscode.workspace.getConfiguration("branchwarnings");
     protectedBranches = config.get<string[]>("protectedBranches");
     suppressPopup = config.get<boolean>("suppressPopup");
-    status.color = config.get<string>("msgColor");
+    status.backgroundColor = new vscode.ThemeColor('statusBarItem.errorBackground');
```
Comment (EEParker) [inline src/extension.ts:50]: This is the primary change, the rest is just scaffolding and npm updates.

File: src/extension.ts [hunk 2]
```diff
@@ -47,7 +47,8 @@ function updateConfigs(status:vscode.StatusBarItem) {
...
+    status.command = "git.checkout";
```
Comment (EEParker) [inline src/extension.ts:51]: This command mimics when you click the `master` branch status bar button to start a branch right from the warning.

**Comment by teledemic** [comment]
This looks great, thanks! Will merge once the API becomes stable.

**Comment by EEParker** [comment]
This is now on the release API for VSCode 1.53. 
https://code.visualstudio.com/updates/v1_53#_extension-authoring

**Comment by teledemic** [comment]
Great, thanks!

**Comment by teledemic** [comment]
I see they redid the vscode testing tool. @EEParker how are you launching a copy of vscode with the extension running for debug?

**Comment by teledemic** [comment]
Never mind, got it working - will publish shortly!

**Comment by EEParker** [comment]
Thanks for getting this published! Getting my whole team on board, this extensions is very useful.

**Pull request merged**